Fix for "Invalid UTF-8 stream" error on valid characters (#476).

vshabanov opened the pull request.

phadej commented:
We definitely should have tests for this, could you add tests from the issue into `aeson`s test suite?

vshabanov commented:
Added a test. Optimized it a bit so it could check all characters in a second instead of half a minute and added basic escapement (like `\n`) test besides UTF-16 `\uXXXX` characters test.

bergmark commented:
Looks great, thanks!
